src/WARP.XrmSolutionAssistant/WARP.XrmSolutionAssistant.Core/SolutionFlowConnectionMapper.cs

```diff
@@ -0,0 +1,247 @@
+
+            return this.GetConnectionReferencesFromCustomizations(customizationsXmlContents);
+        }
+
+        /// <summary>
+        /// Gets a list of connection references from the customizations XElement.
+        /// </summary>
+        /// <param name="customizationsXmlContents">XElement of the customizations.xml file.</param>
+        /// <returns>A List of CustomizationConnectionReferences.</returns>
+        private List<CustomizationsConnectionReference> GetConnectionReferencesFromCustomizations(XElement customizationsXmlContents)
+        {
+            // Get the connection references.
+            var connectionReferencesElement = customizationsXmlContents.Element(ConnectionReferences);
+            var connectionReferenceElements = from el in connectionReferencesElement.Elements(ConnectionReference) select el;
+
+            var connectionReferences = new List<CustomizationsConnectionReference>();
+
+            foreach (var el in connectionReferenceElements)
+            {
+                // Deaerialize each connectionreference element into a class.
+                var ser = new XmlSerializer(typeof(CustomizationsConnectionReference));
+                var conref = (CustomizationsConnectionReference)ser.Deserialize(el.CreateReader());
+                if (conref != null)
+                {
+                    if (connectionReferences.Any(x => x.ApiName == conref.ApiName))
+                    {
+                        // We have already got a connection for this Api. Tag this connection for removal.
+                        Logger.Warn($"Removing connection name [{conref.ConnectionReferenceLogicalName}] from customizations as it is a duplicate for api [{conref.ApiName}].");
+                        this.customizationConnectionsToDelete.Add(conref.ConnectionReferenceLogicalName);
+                        continue;
+                    }
+
+                    connectionReferences.Add(conref);
+                }
+            }
+
+            return connectionReferences;
+        }
+
+        private void ProcessFlowJsons()
+        {
+            Logger.Info("Processing json files.");
+            var workflowsDirectory = Path.Combine(this.solutionRootDirectory, "Workflows");
+            var flowJsonFiles = Directory.GetFiles(workflowsDirectory, "*.json", SearchOption.TopDirectoryOnly);
+
+            foreach (var jsonFile in flowJsonFiles)
+            {
+                Logger.Trace($"Reading {jsonFile}");
+                var jsonContents = File.ReadAllText(jsonFile);
+                foreach (var rep in this.replacements)
+                {
+                    Logger.Trace($"Replacing {rep.Key} with {rep.Value}");
+                    jsonContents = jsonContents.Replace(rep.Key, rep.Value);
+
+                    // Write the updated json file.
+                    Logger.Info("Saving json file.");
+                    var sw = new StreamWriter(jsonFile, false, new UTF8Encoding(true));
```

Comment:
Does this write the same file multiple times if there are multiple replacements required?